tests/unit/types/arrays/test_neural_ops.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+import copy
+
+import numpy as np
+
+from jina.types.arrays.neural_ops import (
+    cosine_distance,
+    euclidean_distance_squared,
+)
+from jina.types.arrays.memmap import DocumentArrayMemmap
+from jina import Document, DocumentArray
+
+
+@pytest.fixture
+def docarrays_for_embedding_distance_computation():
+    d1 = Document(embedding=np.array([0, 0, 0]))
+    d2 = Document(embedding=np.array([3, 0, 0]))
+    d3 = Document(embedding=np.array([1, 0, 0]))
+    d4 = Document(embedding=np.array([2, 0, 0]))
+
+    d1_m = Document(embedding=np.array([1, 0, 0]))
+    d2_m = Document(embedding=np.array([2, 0, 0]))
+    d3_m = Document(embedding=np.array([0, 0, 1]))
+    d4_m = Document(embedding=np.array([0, 0, 2]))
+    d5_m = Document(embedding=np.array([0, 0, 3]))
+
+    D1 = DocumentArray([d1, d2, d3, d4])
+    D2 = DocumentArray([d1_m, d2_m, d3_m, d4_m, d5_m])
+    return D1, D2
+
+
+@pytest.fixture
+def embeddings():
+    return np.array([[1, 0, 0], [2, 0, 0], [3, 0, 0]])
+
+
+@pytest.fixture
+def embedding_query():
+    return np.array([[1, 0, 0]])
+
+
+def test_new_distances_equal_previous_distances():
```

Comment:
I would still propose that this be replaced by comparison against [scipy.distance](https://docs.scipy.org/doc/scipy/reference/spatial.distance.html) module. The dependency is already installed at test time.

Right now it might make sense to compare against "previous distances", but by jina 3.0 nobody will know what these are and why we need to compare against them. However, your call.

Reply:
Will add the test right away and merge then